mozci/console/commands/push.py

```diff
@@ -30,16 +32,78 @@ class ClassifyCommand(Command):
 
     classify
         {branch=mozilla-central : Branch the push belongs to (e.g autoland, try, etc).}
+        {--medium-confidence=0.8 : Medium confidence threshold used to classify the regressions.}
+        {--high-confidence=0.9 : High confidence threshold used to classify the regressions.}
+        {--output= : Path towards a directory to save a JSON file containing classification and regressions details in.}
     """
 
     def handle(self):
+        try:
+            medium_conf = float(self.option("medium-confidence"))
+        except ValueError:
+            self.line("<error>Provided --medium-confidence should be a float.</error>")
+            return
+        try:
+            high_conf = float(self.option("high-confidence"))
+        except ValueError:
+            self.line("<error>Provided --high-confidence should be a float.</error>")
+            return
+
+        output = self.option("output")
+        if output and not os.path.isdir(output):
+            os.makedirs(output)
+            self.line(
+                "<comment>Provided --output pointed to a inexistent directory that is now created.</comment>"
+            )
+
         push = Push(self.argument("rev"), self.argument("branch"))
-        classification = push.classify()
+        classification, regressions = push.classify(
+            confidence_medium=medium_conf, confidence_high=high_conf
+        )
         self.line(
             f'Push associated with the head revision {self.argument("rev")} on the branch '
             f'{self.argument("branch")} is classified as {classification.name}'
         )
 
+        if output:
+            to_save = {
+                "push": {
+                    "id": push.push_uuid,
+                    "classification": classification.name,
+                },
+                "failures": {
+                    "real": {
+                        group: [
+                            {"task_id": task.id, "label": task.label}
+                            for task in failing_tasks
+                        ]
+                        for group, failing_tasks in regressions.real.items()
+                    },
+                    "intermittent": {
+                        group: [
+                            {"task_id": task.id, "label": task.label}
+                            for task in failing_tasks
+                        ]
+                        for group, failing_tasks in regressions.intermittent.items()
+                    },
+                    "unknown": {
+                        group: [
+                            {"task_id": task.id, "label": task.label}
+                            for task in failing_tasks
+                        ]
+                        for group, failing_tasks in regressions.unknown.items()
+                    },
+                },
+            }
+
+            filename = f"{output}/classify_output.json"
```

Comment:
Could you put the branch and revision in the file name? This way when we classify multiple pushes they won't overwrite each other.

Reply:
Sure